bigtable/client/row_reader_test.cc

```diff
@@ -159,6 +159,7 @@ class RowReaderTest : public bigtable::testing::TableTestFixture {
 TEST_F(RowReaderTest, EmptyReaderHasNoRows) {
   EXPECT_CALL(*bigtable_stub_, ReadRowsRaw(_, _)).WillOnce(Return(stream_));
   EXPECT_CALL(*stream_, Read(_)).WillOnce(Return(false));
+  EXPECT_CALL(*stream_, Finish()).WillOnce(Return(grpc::Status::OK));
```

Comment:
What happens (in general) if client code does _not_ call Finish? Couldn't find an answer to this in the gRPC documentation:
https://grpc.io/grpc/cpp/classgrpc_1_1internal_1_1_client_streaming_interface.html
but if the answer is: a call is left dangling, then I should take care of this in the RowReader

Reply:
The gRPC documentation is awful about these things.  I would expect that not calling `Finish()` leaks whatever resources are needed to return the result of `Finish()` when it is eventually called, right? 

On a separate (but related note) I have observed assertion errors for any streaming gRPC that does not follow exactly the right sequence of calls:

https://github.com/GoogleCloudPlatform/cpp-docs-samples/issues/16

In short, I think if we step outside the **exact** sequence of calls that gRPC expects we are wondering into undefined behavior territory.